keycloak-rest-api-extensions-tests/src/test/java/io/cloudtrust/keycloak/credential/CtPasswordCredentialProviderTest.java

```diff
@@ -1,67 +1,84 @@
-import org.keycloak.representations.idm.AdminEventRepresentation;
-import org.keycloak.representations.idm.UserRepresentation;
-
+import io.cloudtrust.keycloak.config.AccountOauthClientConfig;
+import io.cloudtrust.keycloak.config.ServerConfig;
+import io.cloudtrust.keycloak.config.TestRealmConfig;
+import io.cloudtrust.keycloak.test.AbstractKeycloakTest;
+import io.cloudtrust.keycloak.test.ExtensionApi;
 import java.io.IOException;
 import java.net.URISyntaxException;
 import java.util.Map;
+import org.jboss.logging.Logger;
+import org.junit.jupiter.api.Test;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.testframework.annotations.InjectAdminClient;
+import org.keycloak.testframework.annotations.InjectEvents;
+import org.keycloak.testframework.annotations.InjectRealm;
+import org.keycloak.testframework.annotations.KeycloakIntegrationTest;
+import org.keycloak.testframework.events.Events;
+import org.keycloak.testframework.oauth.OAuthClient;
+import org.keycloak.testframework.oauth.annotations.InjectOAuthClient;
+import org.keycloak.testframework.realm.ManagedRealm;
 
-import static io.cloudtrust.keycloak.credential.CtPasswordCreatedAdminEventMatcher.isPasswordCreatedAdminEvent;
+import static io.cloudtrust.keycloak.credential.CtLoginEventMatcher.isLoginEvent;
 import static io.cloudtrust.keycloak.credential.CtPasswordCreatedEventMatcher.isPasswordCreatedEvent;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.not;
 import static org.hamcrest.Matchers.nullValue;
 
-@ExtendWith(KeycloakDeploy.class)
-class CtPasswordCredentialProviderTest extends AbstractRestApiExtensionTest {
+@KeycloakIntegrationTest(config = ServerConfig.class)
+class CtPasswordCredentialProviderTest extends AbstractKeycloakTest {
+
     private static final Logger logger = Logger.getLogger(CtPasswordCredentialProviderTest.class);
 
-    private AdminEventRepresentation pollIt() {
-        AdminEventRepresentation res = adminEvents().poll();
-        return res;
-    }
+    @InjectAdminClient
+    Keycloak keycloak;
 
-    @Test
-    void adminResetPassword() throws IOException, URISyntaxException {
-        RealmResource testRealm = this.getRealm();
-        UserRepresentation user = testRealm.users().search("test-user@localhost").get(0);
+    @InjectRealm(config = TestRealmConfig.class, ref = "test")
+    ManagedRealm testRealm;
 
-        Map<String, Object> body = Map.of(
-                "type", "password",
-                "value", "P@ssw0rd",
-                "temporary", false);
-        this.api().callJSON("PUT", "/admin/realms/test/users/" + user.getId() + "/reset-password", body);
-        assertThat(pollIt(), isPasswordCreatedAdminEvent());
+    @InjectEvents(realmRef = "test")
+    Events events;
+
+    @InjectOAuthClient(realmRef = "test", config = AccountOauthClientConfig.class)
+    OAuthClient oauthClient;
+
+    // TODO to put in parent cloudtrust public
```

Comment:
Remove this... you can use cloudtrust-parent SNAPSHOT version as long as it is not merged/released